components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -1743,6 +1752,41 @@ private void parseSupportedGrantTypesConfig(OMElement oauthConfigElem) {
         }
     }
 
+    private void parseUserConsentEnabledGrantTypesConfig(OMElement oauthConfigElem) {
+
+        OMElement userConsentEnabledGrantTypesElement =
+                oauthConfigElem.getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.USER_CONSENT_ENABLED_GRANT_TYPES));
+
+        if (userConsentEnabledGrantTypesElement != null) {
+            Iterator iterator = userConsentEnabledGrantTypesElement
+                    .getChildrenWithName(getQNameWithIdentityNS(ConfigElements.USER_CONSENT_ENABLED_GRANT_TYPE));
+
+            while (iterator.hasNext()) {
+                OMElement supportedGrantTypeElement = (OMElement) iterator.next();
+                OMElement grantTypeNameElement = supportedGrantTypeElement
+                        .getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.USER_CONSENT_ENABLED_GRANT_TYPE_NAME));
+                String grantTypeName = null;
+                if (grantTypeNameElement != null) {
+                    grantTypeName = grantTypeNameElement.getText();
+                }
+
+                if (StringUtils.isNotEmpty(grantTypeName)) {
+                    userConsentEnabledGrantTypes.add(grantTypeName);
+                } else {
+                    log.warn("Grant Type: " + grantTypeName + " is not a supported grant type. Therefore skipping it " +
```

Comment:
This is a bit misleading isn't it ? Since we are only checking for empty

Reply:
Agreed.